runtime/core/src/main/java/io/atlasmap/core/PathUtil.java

```diff
@@ -1,483 +0,0 @@
-            }
-        }
-
-        return null;
-    }
-
-    public String getCollectionSegment() {
-        for (String part : getSegments()) {
-            if (PathUtil.isCollectionSegment(part)) {
-                return part;
-            }
-        }
-        return null;
-    }
-
-    public void setCollectionIndex(String segment, Integer index) {
-        if (segment == null) {
-            throw new IllegalArgumentException("PathUtil segment cannot be null");
-        }
-
-        if (index < 0) {
-            throw new IllegalArgumentException("PathUtil index must be a positive integer");
-        }
-
-        if (segment.contains(PATH_ARRAY_START) && segment.contains(PATH_ARRAY_END)) {
-            for (int i = 0; i < getSegments().size(); i++) {
-                String part = cleanPathSegment(getSegments().get(i));
-                if (part != null && part.equals(cleanPathSegment(segment))) {
-                    getSegments().set(i, cleanPathSegment(segment) + PATH_ARRAY_START + index + PATH_ARRAY_END);
-                }
-            }
-        } else if (segment.contains(PATH_LIST_START) && segment.contains(PATH_LIST_END)) {
-            for (int i = 0; i < getSegments().size(); i++) {
-                String part = cleanPathSegment(getSegments().get(i));
-                if (part != null && part.equals(cleanPathSegment(segment))) {
-                    getSegments().set(i, cleanPathSegment(segment) + PATH_LIST_START + index + PATH_LIST_END);
-                }
-            }
-        } else {
-            throw new IllegalArgumentException("PathUtil segment is not a List or Array segment");
-        }
-    }
-
-    public String toString() {
-        StringBuilder builder = new StringBuilder();
-
-        int i = 0;
-
-        if (getSegments().size() > 1) {
-            builder.append(PATH_SEPARATOR);
-        }
-
-        for (String part : getSegments()) {
-            builder.append(part);
-            if (i < (getSegments().size() - 1)) {
-                builder.append(PATH_SEPARATOR);
-            }
-            i++;
-        }
-        return builder.toString();
-    }
-
-    public String getOriginalPath() {
-        return originalPath;
-    }
-
-    public static boolean isArraySegment(String segment) {
-        return isCollectionSegment(segment) && segment.contains(PATH_ARRAY_START);
-    }
-
-    public static boolean isListSegment(String segment) {
-        return isCollectionSegment(segment) && segment.contains(PATH_LIST_START);
-    }
-
-    public static boolean isMapSegment(String segment) {
-        return isCollectionSegment(segment) && segment.contains(PATH_MAP_START);
-    }
-
-    public static String findNextSegment(String segment, String path) {
-        PathUtil pathUtil = new PathUtil(path);
-        String parentSegment = null;
-        for (String currentSegment : pathUtil.segments) {
-            if (parentSegment == null || !segment.equals(parentSegment)) {
```

Comment:
Yeah I think Codacy is right here, this loop looks super weird. @sunilmeddr I know it was just copied, but can you turn over the condition and just return in 'if'?
https://www.codacy.com/app/atlasmapio/atlasmap/pullRequest?prid=1131462